logger library should not be required by clients

The logger library (src/logger) gets built as a static library `acquire-logger`. 

When someone wants to depend on the `acquire-zarr` static library, they need to also link `acquire-logger`. But that library is only providing definitions for private symbols in `acquire-zarr` - it's not required for any part of that library's public interface.

Here's a proposed approach:

Ideally, `acquire-zarr` would be built in such a way that all the `acquire-logger` symbol definitions are all included. We could do this by listing the source files in src/logger directly or maybe by exposing on object library target. I believe cmake effectively adds the object files when compiling the lib which would resolve the issue.

in case it's at all useful, if you want to see what I was trying to do when I cam across this error:
install meson, ninja, and [just](https://github.com/casey/just)

```sh
git clone https://github.com/tlambert03/pymmcore-nano.git --branch storage-device --depth=1
cd pymmcore-nano
just build-adapter Go2Scope
```

<details>

<summary>link error</summary>

```
[193/193] Linking target libmmgr_dal_Go2Scope.dylib
FAILED: libmmgr_dal_Go2Scope.dylib
c++  -o libmmgr_dal_Go2Scope.dylib libmmgr_dal_Go2Scope.dylib.p/AcqZarrStorage.cpp.o libmmgr_dal_Go2Scope.dylib.p/CFileUtil.cpp.o libmmgr_dal_Go2Scope.dylib.p/G2SStorage.cpp.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -shared -install_name @rpath/libmmgr_dal_Go2Scope.dylib subprojects/MMDevice/libMMDevice.a subprojects/acquire-zarr/libacquire_zarr.a /opt/homebrew/Cellar/boost/1.86.0_2/lib/libboost_filesystem-mt.dylib /opt/homebrew/Cellar/boost/1.86.0_2/lib/libboost_system-mt.dylib /Users/talley/dev/forks/vcpkg/installed/arm64-osx/debug/lib/libblosc.a /Users/talley/dev/forks/vcpkg/installed/arm64-osx/debug/lib/libminiocpp.a /Users/talley/dev/forks/vcpkg/installed/arm64-osx/lib/libz.a /Users/talley/dev/forks/vcpkg/installed/arm64-osx/debug/lib/libzstd.a /Users/talley/dev/forks/vcpkg/installed/arm64-osx/debug/lib/libcurlpp.a /Users/talley/dev/forks/vcpkg/installed/arm64-osx/debug/lib/libINIReader.a /Users/talley/dev/forks/vcpkg/installed/arm64-osx/lib/libssl.a /Users/talley/dev/forks/vcpkg/installed/arm64-osx/lib/libcrypto.a /Users/talley/dev/forks/vcpkg/installed/arm64-osx/debug/lib/liblz4d.a /Users/talley/dev/forks/vcpkg/installed/arm64-osx/debug/lib/libinih.a /Users/talley/dev/forks/vcpkg/installed/arm64-osx/debug/lib/libpugixml.a -lcurl
Undefined symbols for architecture arm64:
  "Logger::log_mutex_", referenced from:
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [26], char const*>(LogLevel, char const*, int, char const*, char const (&) [26], char const*&&) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [23]>(LogLevel, char const*, int, char const*, char const (&) [23]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [26], unsigned long&>(LogLevel, char const*, int, char const*, char const (&) [26], unsigned long&) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [41]>(LogLevel, char const*, int, char const*, char const (&) [41]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [29], char const*>(LogLevel, char const*, int, char const*, char const (&) [29], char const*&&) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [42]>(LogLevel, char const*, int, char const*, char const (&) [42]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [21]>(LogLevel, char const*, int, char const*, char const (&) [21]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      ...
  "Logger::get_log_level()", referenced from:
      _Zarr_get_log_level in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
  "Logger::set_log_level(LogLevel)", referenced from:
      _Zarr_set_log_level in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
  "Logger::current_level_", referenced from:
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [26], char const*>(LogLevel, char const*, int, char const*, char const (&) [26], char const*&&) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [23]>(LogLevel, char const*, int, char const*, char const (&) [23]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [26], unsigned long&>(LogLevel, char const*, int, char const*, char const (&) [26], unsigned long&) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [41]>(LogLevel, char const*, int, char const*, char const (&) [41]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [29], char const*>(LogLevel, char const*, int, char const*, char const (&) [29], char const*&&) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [42]>(LogLevel, char const*, int, char const*, char const (&) [42]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [21]>(LogLevel, char const*, int, char const*, char const (&) [21]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      ...
  "Logger::get_timestamp_()", referenced from:
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [26], char const*>(LogLevel, char const*, int, char const*, char const (&) [26], char const*&&) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [23]>(LogLevel, char const*, int, char const*, char const (&) [23]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [26], unsigned long&>(LogLevel, char const*, int, char const*, char const (&) [26], unsigned long&) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [41]>(LogLevel, char const*, int, char const*, char const (&) [41]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [29], char const*>(LogLevel, char const*, int, char const*, char const (&) [29], char const*&&) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [42]>(LogLevel, char const*, int, char const*, char const (&) [42]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> Logger::log<char const (&) [21]>(LogLevel, char const*, int, char const*, char const (&) [21]) in libacquire_zarr.a[2](src_streaming_acquire.zarr.cpp.o)
      ...
ld: symbol(s) not found for architecture arm64
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

</details>

and here is the contents of the relevant wrap file at `src/mmCoreAndDevices/DeviceAdapters/Go2Scope/subprojects/acquire-zarr.wrap`


```ini
[wrap-git]
url = https://github.com/acquire-project/acquire-zarr
revision = v0.0.2
depth = 1
method = cmake

[provide]
acquire-zarr = acquire_zarr_dep
```

--- a/benchmarks/CMakeLists.txt
+++ b/benchmarks/CMakeLists.txt
@@ -11,7 +11,6 @@ target_include_directories(${tgt} PRIVATE
         ${PROJECT_SOURCE_DIR}/src/logger
 )
 target_link_libraries(${tgt} PRIVATE
-        acquire-logger
         acquire-zarr
         nlohmann_json::nlohmann_json
         miniocpp::miniocpp

--- a/src/logger/CMakeLists.txt
+++ b/src/logger/CMakeLists.txt
@@ -1,19 +1,14 @@
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-
-add_library(acquire-logger
+add_library(acquire-logger-obj OBJECT
         logger.hh
         logger.cpp
 )
 
-target_include_directories(acquire-logger
-        INTERFACE
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-)
-
-set_target_properties(acquire-logger PROPERTIES
+set_target_properties(acquire-logger-obj PROPERTIES
+        POSITION_INDEPENDENT_CODE ON
         MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>"
 )
 
-install(TARGETS acquire-logger
-        LIBRARY DESTINATION lib
+target_include_directories(acquire-logger-obj
+        PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
 )

--- a/src/streaming/CMakeLists.txt
+++ b/src/streaming/CMakeLists.txt
@@ -31,6 +31,7 @@ add_library(${tgt}
         zarrv3.array.writer.cpp
         vectorized.file.writer.hh
         vectorized.file.writer.cpp
+        $<TARGET_OBJECTS:acquire-logger-obj>
 )
 
 target_include_directories(${tgt}
@@ -42,7 +43,6 @@ target_include_directories(${tgt}
 )
 
 target_link_libraries(${tgt} PRIVATE
-        acquire-logger
         blosc_static
         miniocpp::miniocpp
 )

--- a/tests/integration/CMakeLists.txt
+++ b/tests/integration/CMakeLists.txt
@@ -23,7 +23,6 @@ foreach (name ${tests})
             ${PROJECT_SOURCE_DIR}/src/logger
     )
     target_link_libraries(${tgt} PRIVATE
-            acquire-logger
             acquire-zarr
             nlohmann_json::nlohmann_json
             miniocpp::miniocpp

--- a/tests/unit-tests/CMakeLists.txt
+++ b/tests/unit-tests/CMakeLists.txt
@@ -41,7 +41,6 @@ foreach (name ${tests})
             ${PROJECT_SOURCE_DIR}/src/streaming
     )
     target_link_libraries(${tgt} PRIVATE
-            acquire-logger
             acquire-zarr
             miniocpp::miniocpp
     )

